pkg/library/lifecycle/lifecycle.go

```diff
@@ -71,6 +73,50 @@ func GetInitContainers(devfile v1alpha2.DevWorkspaceTemplateSpecContent) (initCo
 	return initContainers, mainComponents, nil
 }
 
+func ApplyPostStartCommands(devfile *v1alpha2.DevWorkspaceTemplateSpecContent, containers []corev1.Container) error {
+	postStartCommands, err := getCommandsForKeys(devfile.Events.PostStart, devfile.Commands)
+	if err != nil {
+		return fmt.Errorf("failed to process preStart commands: %w", err)
+	}
+	for _, cmd := range postStartCommands {
+		if cmd.Exec == nil {
+			return fmt.Errorf("problem processing postStart event %s: only exec-type commands are supported", cmd.Key())
+		}
+		container, err := getContainerByName(cmd.Exec.Component, containers)
+		if err != nil {
+			return fmt.Errorf("failed to find container for command %s", cmd.Key())
+		}
+		err = writePostStartLifecycle(container, cmd.Exec.CommandLine)
+		if err != nil {
+			return fmt.Errorf("error applying postStart lifecycle hook to container: %w", err)
+		}
+	}
+	return nil
+}
+
+func writePostStartLifecycle(container *corev1.Container, cmd string) error {
+	if container.Lifecycle == nil {
+		container.Lifecycle = &corev1.Lifecycle{}
+	}
+	if container.Lifecycle.PostStart == nil {
+		container.Lifecycle.PostStart = &corev1.Handler{
+			Exec: &corev1.ExecAction{
+				Command: []string{
```

Comment:
According to https://kubernetes.io/docs/tasks/configure-pod-container/attach-handler-lifecycle-event/#discussion

> The Container's status is not set to RUNNING until the postStart handler completes.

which means, it can't serve the traffic because service just won't pick such pod up.

Install vsx should be relatively quick operation but depends on the network.
Also, I see other postStart operations which developer may want to start on workspace start to speed up processes - like make yarn, mvn, go mod download dependencies. And the editor should be able to listen to output or result of such operations.

So, I'm not sure who should manage postStart commands, editor or DevWorkspace controller.
If DevWorkspace controller, it makes sense to do it out from K8s API and add API to follow command status/output.

Reply:
Makes sense.